reference/gall.md

```diff
@@ -0,0 +1,765 @@
+particular path.  The agent may or may not need to perform setup steps
+to intialize the subscription.  It may produce a `%give`
+`%subscription-result` to the subscriber to get it up to date, but after
+this event is complete, it cannot give further updates to a specific
+subscriber.  It must give all further updates to all subscribers on a
+specific path.
+
+If this arm crashes, then the subscription is immediately terminated.
+More specifcally, it never started -- the subscriber will receive a
+negative `%watch-ack`.  You may also produce an explicit `%kick` to
+close the subscription without crashing -- for example, you could
+produce a single update followed by a `%kick`.
+
+#### Accepts
+
+```hoon
+=path
+```
+
+Path of new subscription.
+
+#### Returns
+
+```hoon
+(quip card _this)
+```
+
+List of cards and new agent.
+
+#### Example
+
+```hoon
+:: From reference agent. Gives new subscribers to /local the current local counter.
+++  on-watch
+  |=  =path
+  ^-  (quip card _this)
+  :_  this
+  ~&  path
+  ?+  path  ~
+       ::
+     [%local ~]
+  [%give %fact ~ %atom !>(local.sta)]~
+  ==
+```
+
+### `++on-leave`
+
+Handle unsubscribe.
+
+This arm is called when a program becomes unsubscribed to you.
+Subscriptions may close because the subscriber intentionally
+unsubscribed, but they also could be closed by an intermediary.  For
+example, if a subscription is from another ship which is currently
+unreachable, Ames may choose to close the subscription to avoid queueing
+updates indefinitely.  If the program crashes while processing an
+update, this may also generate an unsubscription.  You should consider
+subscriptions to be closable at any time.
+
+#### Accepts
+
+```hoon
+=path
+```
+
+Path of the closed subscription.
+
+#### Returns
+
+```hoon
+(quip card _this)
+```
+
+List of cards and new agent.
+
+#### Example
+
+```hoon
+:: From reference agent. Prints a message when programs become unsubscribed.
+++  on-leave
+  |=  =path
+  ^-  (quip card _this)
+  ~&  "Unsubscribe by: {<src.bowl>} on: {<path>}"
+  `this
+```
+
+### `++on-peek`
+
+Handle scry request.
+
+This arm is called when a program reads from the agent's "scry"
+namespace, which should be referentially transparent.  Unlike most
+handlers, this cannot perform IO, and it cannot change the state.  All
+it can do is produce a piece of data to the caller, or not.
+
+#### Accepts
+
+```hoon
+=path
+```
+
+The path being scryed for.
+
+#### Returns
+
+```hoon
+(unit (unit cage))
+```
+
+If this arm produces `[~ ~ data]`, then `data` is the value at the the
+given path.  If it produces `[~ ~]`, then there is no data at the given
+path and never will be.  If it produces `~`, then we don't know yet whether
+there is or will be data at the given path. The head of the path is known as the
+`care`. Requests with a care of `%x` should return a vase that matches the mark
+at the end of the path. Requests with a care of `%y` should return a vase with
+an `arch` inside.
```

Comment:
Perhaps also worth noting is that, while the mark that's being asked for isn't included in the `path` that you're given, your output `cage` needs to match it (or be convertible to it?) in order for it to not crash on the `.^` end.